xclim/indices/run_length.py

```diff
@@ -465,16 +470,20 @@ def get_out(d):
         if freq is not None:
             out = da.resample({dim: freq}).map(get_out)
         else:
-            out = xr.where(da.any(dim=dim), da.argmax(dim=dim), np.NaN)
+            out = get_out(da)
             out = coord_transform(out, da)
 
+    # Last_run not working in this state
     elif ufunc_1dim:
+        if first is False:
+            da = da[{dim: slice(None, None, -1)}]
         out = first_run_ufunc(x=da, window=window, dim=dim)
+        if first is False:
+            out = da[dim].size - out - 2
         out = coord_transform(out, da)
```

Comment:
I just realized passing `index="last"` to `first_run_ufunc` should do the trick to make it work too, will test this soon